### PR TITLE
update internals CHANGELOG for release

### PR DESCRIPTION
**Pull request by apoelstra**
.

### CODE REVIEW

**Review by Kixunil — DISMISSED** [review]
ACK be02045561aec320c8d929c7a2c66511dfc5f058

But do we even need changelog for internals? People aren't supposed to use it.

**Review by sanket1729 — DISMISSED** [review]
utACk be02045561aec320c8d929c7a2c66511dfc5f058

**Comment by apoelstra** [comment]
> But do we even need changelog for internals? People aren't supposed to use it.

Heh, good point. I think it's worthwhile to have it for our own sakes.

**Comment by apoelstra** [comment]
Thanks for the ACKs guys, but I remembered as I was falling asleep that I forgot an important part of this PR -- I have to specify an explicit `version` in the Cargo.toml for `bitcoin_hashes` (and for `bitcoin`). Crates will not let me publish crates unless all their deps have explicit versions.

With this change, if we merge this then we can release `hashes` and `internals` from the same commit. I'm pretty sure.

**Comment by Kixunil** [comment]
Yeah, IIRC sufficiently recent cargo allows this.

**Review by Kixunil — DISMISSED** [review]
ACK ca3b4330e8c720428ccc81622c185809b9eb807e

**Review by tcharding — COMMENTED** [review]
ACK ca3b4330e8c720428ccc81622c185809b9eb807e

**Review by tcharding — DISMISSED** [review]

**Comment by apoelstra** [comment]
Moving to draft since `bitcoin-internals = "0.1.0"` currently points to a malicious crate (currently, an unformatted version of our own, but that could change). See #1702.

**Comment by apoelstra** [comment]
Ok, renamed the internal crate to bitcoin-private since we have gotten no response from the guy who is squatting on bitcoin-internals. This PR should be good to go now.

File: internals/Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [package]
-name = "bitcoin-internals"
+name = "bitcoin-private"
 version = "0.1.0"
 authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>", "The Rust Bitcoin developers"]
 license = "CC0-1.0"
 repository = "https://github.com/rust-bitcoin/rust-bitcoin/"
-documentation = "https://docs.rs/bitcoin-internals"
+documentation = "https://docs.rs/bitcoin-private"
```
Comment (Kixunil) [inline internals/Cargo.toml:7]: I would drop this field entirely since it's the default anyway. Non-blocking.

**Review by Kixunil — APPROVED** [review]
ACK 51f2f8483ccef7c73e4e9e643ad0dd4406891aa7

**Review by tcharding — APPROVED** [review]
ACK 51f2f8483ccef7c73e4e9e643ad0dd4406891aa7

**Comment by apoelstra** [comment]
Tagged and published `bitcoin-private` 0.1.0 and `bitcoin_hashes` 0.12.0.

**Pull request merged**